File: test/boost/ExampleTest.cpp
#define BOOST_TEST_DYN_LINK
#define BOOST_TEST_MODULE ExampleTest
#include <boost/test/unit_test.hpp>

BOOST_AUTO_TEST_CASE(testSuccess)
{
    BOOST_CHECK(true);
}

BOOST_AUTO_TEST_CASE(testFailure)
{
    BOOST_CHECK(false);
}

File: test/junit/ExampleTest.java
import org.junit.Test;
import static org.junit.Assert.assertTrue;

public class ExampleTest {

    @Test
    public void testSuccess() {
        assertTrue(true);
    }

    @Test
    public void testFailure() {
        assertTrue(false);
    }

    @Test
    public void testError() {
        Object o = null;
        o.hashCode();
    }

}
